docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md

```diff
@@ -0,0 +1,177 @@
+
+```mermaid
+classDiagram
+ class MonthlyMatchesByAlgorithm
+    MonthlyMatchesByAlgorithm: +Integer id
+    MonthlyMatchesByAlgorithm: +Date month 
+    MonthlyMatchesByAlgorithm: +Integer doi
+    MonthlyMatchesByAlgorithm: +Integer issn
+    MonthlyMatchesByAlgorithm: +Integer isbn
+    MonthlyMatchesByAlgorithm: +Integer pmid
+    MonthlyMatchesByAlgorithm: +Integer journal_title
+    MonthlyMatchesByAlgorithm: +Integer staff_categorized
+    MonthlyMatchesByAlgorithm: +Integer no_matches
+```
+
+This approach is likely complementary to the "count all matches" approach and as both store minimal data it is possible we should do both. It is likely a single pass through all Terms could populate both tables if a lookup were done to see if the Term had a SearchEvent. Care would need to be taken to not create an N+1 SQL query if doing this in a single job but it should be possible.
+
+#### Detections table with verbose data about matches
+
+A more verbose way to record detections from our algorithms might be a log-like table that ties to SearchEvents.
+
+##### Class diagram for verbose Detections Table
+
+```mermaid
+classDiagram
+
+  Term --< SearchEvent : has many
+  SearchEvent --< SearchDetections : has many
+
+  class Term
+    Term: +Integer id
+    Term: +String phrase
+
+  class SearchEvent
+    SearchEvent: +Intger id
+    SearchEvent: +Integer term_id
+    SearchEvent: +Timestamp created_at
+
+  class SearchDetections
+    SearchDetections: +Integer id
+    SearchDetections: +Integer searchevent_id
+    SearchDetections: +JSON DOI NULL
+    SearchDetections: +JSON ISBN NULL
+    SearchDetections: +JSON ISSN NULL
+    SearchDetections: +JSON PMID NULL
+    SearchDetections: +JSON Hint NULL
+    SearchDetections: +Timestamp created_at
+```
+
+We already have a record of every time we’ve seen a given search for any phrase - adding a table (or even fields in the SearchEvent table) for each detector we build would allow us to have a record of whether each detector fired, and what it returned. Setting a default value of NULL on each field, but storing an empty non-NULL value via code, would allow us to query for failures to detect (which would be stored as {} separately from any search prior to that detector being in place (which would be stored as NULL values).
+
+An approach like this, which doesn’t generate only counts but also the details of what is detected / returned, would provide greater visibility into application performance - but at the cost of needing to be recorded at query time, and not as a batch retroactively.
+
+This stores a lot more data and it isn't clear what we would use it for. This approach should likely be implemented in the future if we find we have a specific question that can only be solved with this level of detail. We would lose historical data if we wait to implement this, but that risk is outweighed by avoiding storing a bunch of data without a clear purpose.
```

Comment:
Alternatively, we could store the data for now and drop the table if it becomes clear that there's no purpose for it. The downside of that (aside from storing a lot of data in the interim) is knowing _when_ to decide whether it's useful or not.

Reply:
I concur - knowing when to get rid of that data feels like a thornier process than just not collecting it yet until we have a more clear understanding of how we'll handle it and when to delete it.